### PR TITLE
Fix npm publish

### PR DESCRIPTION
**Pull request by oeway**
Rollback to simpler publish script due to a bug in the previous one.

### CODE REVIEW

File: .github/workflows/build-library.yml [hunk 1]
```diff
@@ -71,14 +71,8 @@ jobs:
...
-        uses: pascalgn/npm-publish-action@1.1.0
-        with:
-          tag_name: "v%s"
-          tag_message: "v%s"
-          commit_pattern: "^Release (\\S+)"
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - run: npm ci
+      - run: npm build
```
Comment (MartinHjelmare) [inline .github/workflows/build-library.yml]: Do we need to build besides installing with `npm ci`?

Comment (MartinHjelmare) [reply .github/workflows/build-library.yml]: Yes, we do. Install, just installs the dependencies. We need to run the build actions in the build script.

File: .github/workflows/build-library.yml [hunk 2]
```diff
@@ -71,14 +71,8 @@ jobs:
...
+      - name: Try to publish to npm
+        run: npm publish --access public
         continue-on-error: true
```
Comment (MartinHjelmare) [inline .github/workflows/build-library.yml]: Why do we need this?

File: .github/workflows/build-library.yml (hunk 1 above)
Comment (oeway) [reply .github/workflows/build-library.yml]: Yes, we need to generate the `dist` folder for publishing.

Comment (MartinHjelmare) [reply .github/workflows/build-library.yml]: Although we could download the artifact from the build job instead and save some time.

File: .github/workflows/build-library.yml (hunk 2 above)
Comment (oeway) [reply .github/workflows/build-library.yml]: We are not changing the version every time, sometimes we do a PR just for updating README, and that will result in failure of publishing, then we will see a cross with the merged commit message.

**Comment by oeway** [comment]
@MartinHjelmare could you check if there is anything to change, I can only test the changes for publishing after merging.

**Comment by MartinHjelmare** [comment]
See my two comments above.

File: .github/workflows/build-library.yml (hunk 2 above)
Comment (MartinHjelmare) [reply .github/workflows/build-library.yml]: Then the action shouldn't error, I think. Can we change that?

File: .github/workflows/build-library.yml (hunk 1 above)
Comment (oeway) [reply .github/workflows/build-library.yml]: Good idea, then we can even skip the npm install step.

File: .github/workflows/build-library.yml [hunk 3]
```diff
@@ -72,12 +72,15 @@ jobs:
...
       - run: npm install
+      - run: npm build
       - name: Publish if version has been updated
-        uses: pascalgn/npm-publish-action@1.1.0
+        uses: oeway/npm-publish-action@97c098d
         with:
           tag_name: "v%s"
           tag_message: "v%s"
           commit_pattern: "^Release (\\S+)"
+          commit_user: oeway
```
Comment (MartinHjelmare) [inline .github/workflows/build-library.yml]: I think you need to add these new options as inputs in the forked action.

Comment (oeway) [reply .github/workflows/build-library.yml]: What do you mean by `as inputs`?

Comment (MartinHjelmare) [reply .github/workflows/build-library.yml]: https://help.github.com/en/actions/building-actions/metadata-syntax-for-github-actions#inputs

File: .github/workflows/build-library.yml (hunk 2 above)
Comment (MartinHjelmare) [reply .github/workflows/build-library.yml]: Or maybe only run the publish step if the version has changed.

File: .github/workflows/build-library.yml (hunk 3 above)
Comment (oeway) [reply .github/workflows/build-library.yml]: Got it now, just did: https://github.com/oeway/npm-publish-action/commit/2c7c8b50a9d8c19f76059fc5b6b45a27f29df3b0

Thanks.

**Comment by oeway** [comment]
Alright, will merge and give it a try.

**Comment by MartinHjelmare** [comment]
We can address the error comments in another PR.

**Pull request merged**